pkg/ccl/testccl/sqlccl/tenant_gc_test.go

```diff
@@ -130,6 +131,253 @@ func TestGCTenantRemovesSpanConfigs(t *testing.T) {
+		defer mu.Unlock()
+		require.Equal(t, false, mu.isProtected)
+	}
+
+	testutils.RunTrueAndFalse(t, "secondary-tenant", func(t *testing.T, forSecondaryTenant bool) {
+		resetTestingKnob()
+		if !forSecondaryTenant {
+			return
+		}
+		registry := ts.JobRegistry().(*jobs.Registry)
+		execCfg := ts.ExecutorConfig().(sql.ExecutorConfig)
+		sqlRunner := sqlDB
+		if forSecondaryTenant {
+			registry = tt.JobRegistry().(*jobs.Registry)
+			execCfg = tt.ExecutorConfig().(sql.ExecutorConfig)
+			sqlRunner = ttSQLDB
+		}
+		sqlRunner.Exec(t, "CREATE TABLE t()")
+		var tableID int
+		sqlRunner.QueryRow(t, "SELECT id FROM system.namespace WHERE name = 't'").Scan(&tableID)
+		// Drop the table. We'll also create another GC job for this table with
+		// an injected drop time.
+		sqlRunner.Exec(t, "DROP TABLE t")
+
+		// Write a PTS over the entire keyspace. We'll include multiple
+		// timestamps, one of which is before the drop time.
+		r1, err := spanconfig.MakeRecord(
+			spanconfig.MakeTargetFromSystemTarget(spanconfig.MakeEntireKeyspaceTarget()),
+			makeSystemSpanConfig(8, 23, 30),
+		)
+
+		// We do something similar for a PTS set by the host over the tenant's
+		// keyspace.
+		hostOnTenant, err := spanconfig.MakeTenantKeyspaceTarget(roachpb.SystemTenantID, tenantID)
+		require.NoError(t, err)
+		r2, err := spanconfig.MakeRecord(
+			spanconfig.MakeTargetFromSystemTarget(hostOnTenant),
+			makeSystemSpanConfig(6, 28, 35),
+		)
+
+		// Lastly, we'll also add a PTS set by the host over a different
+		// secondary tenant's keyspace. This should have no bearing on our test.
+		hostOnTenant20, err := spanconfig.MakeTenantKeyspaceTarget(
+			roachpb.SystemTenantID, roachpb.MakeTenantID(20),
+		)
+		require.NoError(t, err)
+		r3, err := spanconfig.MakeRecord(
+			spanconfig.MakeTargetFromSystemTarget(hostOnTenant20),
+			makeSystemSpanConfig(3, 4, 5),
+		)
+
+		require.NoError(t, err)
+		err = tsKVAccessor.UpdateSpanConfigRecords(ctx, nil /*toDelete */, []spanconfig.Record{r1, r2, r3})
+		require.NoError(t, err)
+
+		// One more, this time set by the tenant itself on its entire keyspace.
+		// Note that we must do this upsert using the tenant's KVAccessor.
+		tenantOnTenant, err := spanconfig.MakeTenantKeyspaceTarget(tenantID, tenantID)
+		require.NoError(t, err)
+		r4, err := spanconfig.MakeRecord(
+			spanconfig.MakeTargetFromSystemTarget(tenantOnTenant),
+			makeSystemSpanConfig(2, 40),
+		)
+		require.NoError(t, err)
+		err = ttKVAccessor.UpdateSpanConfigRecords(ctx, nil /*toDelete */, []spanconfig.Record{r4})
+		require.NoError(t, err)
+
+		sj, err := jobs.TestingCreateAndStartJob(
+			ctx, registry, execCfg.DB, makeGCJobRecord(10, descpb.ID(tableID)),
+		)
+		require.NoError(t, err)
+
+		ensureGCBlockedByPTS(t, registry, sj)
+
+		// Update PTS state that applies to the entire keyspace -- we only
+		// remove the timestamp before the drop time. We should expect GC to
+		// succeed on if we're not testing for the system tenant at this point.
```

Comment:
This seems a little at odds with the outline at the top of the test. Isn't the expectation that removing the entire keyspace protection policy below the drop time, affects both the system and secondary tenants? i.e. allows GC on both? I think thats what we are checking below bu the last line of this comment is a little misleading.